CaesarCipher.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * To change this license header, choose License Headers in Project Properties.
+ * To change this template file, choose Tools | Templates
+ * and open the template in the editor.
+ */
+package caesarcipher;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ *
+ * @author kailabillie
+ */
+public class CaesarCipher {
+
+    /**
+     * @param args the command line arguments
+     */
+    public static void main(String[] args) {
+
+        String encodedString = Encode("-1:some text");
+        System.out.println(encodedString);
+
+    }
+
+    static String Encode(String word) {
+
+        String uppercase = "ABCDEFGHIJKLMNOPQRSTUVWXYZ";
+        String lowercase = "abcdefghijklmnopqrstuvwxyz";
+        String digits = "1234567890";
+        
+        String [] words = word.split(":");
+        
+        
+        
+        int letterShift = Integer.parseInt(words[0]);
+        String encodeWord = words[1];
+        
+        System.out.println(encodeWord);
+        
+        String newWord = "";
+        
+        for (int i = 0; i<encodeWord.length(); i++) {
+            if (String.valueOf(encodeWord.charAt(i)).equals(" ")) {
+                    newWord += " ";
+                }
+            if (uppercase.contains(String.valueOf(encodeWord.charAt(i)))) { 
+                int index = uppercase.indexOf(encodeWord.charAt(i));
+                int newIndex = (letterShift + index) % 26;
```

Comment:
While running JUnit test cases, the operations on lines 51, 57, and 63 were stopping some scenarios from passing.  newIndex evaluates to a negative number, and throws an exception when trying to access the string array. Try replacing those statements with:
`char letter = newIndex < 0 ? lowercase.charAt(newIndex + 26) : lowercase.charAt(newIndex);`
`char letter = newIndex < 0 ? uppercase.charAt(newIndex + 26) : uppercase.charAt(newIndex);`
`char letter = newIndex < 0 ? digits.charAt(newIndex + 10) : digits.charAt(newIndex);`

 This does the same thing if the index is positive, but wraps the string backwards from A to Z if the index is negative. For example, "-1: AAAAA". Instead of accessing an index out of bounds ( 0 - 1 = -1), it adds 26 to ensure it loops correctly backwards to position 25 ('ZZZZZ').